thunder/torch/__init__.py

```diff
@@ -1761,6 +1761,18 @@ def real(a):
 # TODO Move these to torch.nn.functional
 
 
+@torchsymbol(torch.celu, torch.nn.functional.celu, id="torch.celu", is_method=True)
+def celu(a: TensorLike, /, alpha: float = 1.0, inplace: bool = False) -> TensorLike:
```

Comment:
In the PyTorch reference implementation (https://github.com/pytorch/pytorch/blob/869d1ad0b4d85e0528920949162f1bdcf6536bc6/torch/_refs/nn/functional/__init__.py#L161) `alpha` defaults to `None`:

```
alpha: None | float = None
```

This allows for a special case:

```
if alpha is None:
  rhs = torch.expm1(a)
```

In the future, we might want to add special-cases to division and multiplication so that they would be no-ops, anyway (fyi @kshitij12345, @tfogal).

Reply:
Right, this is a point I do not understand.  There is the reference implementation that you link to, but then there are also these two entry points: [here](https://github.com/pytorch/pytorch/blob/main/torch/nn/modules/activation.py#L568) and [here](https://github.com/pytorch/pytorch/blob/a762dc0357d6c78bbb8be942355cad21b704debe/torch/nn/functional.py#L1849).  Both of these provide a default value of 1.0 for `alpha`.  So which signature should thunder ops be mirroring?

Reply:
I appreciate the efficiency of not doing redundant arithmetic, however I encounter the following error when following the suggestion:
```
input = tensor(3.5244, device='cuda:0', dtype=torch.float64), alpha = None, inplace = False

    def celu(
        input: Tensor,
        alpha: float = 1.0,
        inplace: bool = False,
    ) -> Tensor:  # noqa: D400,D402
        r"""celu(input, alpha=1., inplace=False) -> Tensor

        Applies element-wise,
        :math:`\text{CELU}(x) = \max(0,x) + \min(0, \alpha * (\exp(x/\alpha) - 1))`.

        See :class:`~torch.nn.CELU` for more details.
        """
        if has_torch_function_unary(input):
            return handle_torch_function(
                celu, (input,), input, alpha=alpha, inplace=inplace
            )
        if inplace:
            result = torch.celu_(input, alpha)
        else:
>           result = torch.celu(input, alpha)
E           TypeError: celu(): argument 'alpha' (position 2) must be Number, not NoneType
```

Since PyTorch does not expect `None`, I will leave it as is.